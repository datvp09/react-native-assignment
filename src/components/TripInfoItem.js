import React, { Component } from "react"
import { StyleSheet, Text, View, TouchableWithoutFeedback } from "react-native"
import moment from "moment"

export default class TripInfoItem extends Component {
  onItemPress = () => {
    if (this.props.onPress) {
      this.props.onPress()
    }
  }

  render() {
    const { item, style } = this.props
    const { trip_id, distance, duration, end_time, start_time } = item
    const startTimeText = moment(start_time)
      .utc()
      .format("H:mm, D/M/YYYY")
    const endTimeText = moment(end_time)
      .utc()
      .format("H:mm, D/M/YYYY")

    return (
      <TouchableWithoutFeedback onPress={this.onItemPress}>
        <View style={[styles.rootView, style]}>
          <View style={styles.container}>
            <Text style={styles.text}>{"TripId: " + trip_id}</Text>
            <Text style={styles.text}>{"Distance: " + distance}</Text>
            <Text style={styles.text}>{"Duration: " + duration}</Text>
          </View>
          <View style={styles.containerBig}>
            <Text style={styles.text}>{"Start time: " + startTimeText}</Text>
            <Text style={styles.text}>{"End time: " + endTimeText}</Text>
          </View>
        </View>
      </TouchableWithoutFeedback>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 2
  },
  containerBig: {
    flex: 3
  },
  rootView: {
    backgroundColor: "white",
    borderRadius: 4,
    opacity: 0.9,
    marginHorizontal: 7,
    marginBottom: 10,
    padding: 10,
    paddingBottom: 12,
    flexDirection: "row"
  },
  text: {
    color: "black",
    fontSize: 12
  }
})
