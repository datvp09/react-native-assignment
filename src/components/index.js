import TripInfoItem from './TripInfoItem';

export {TripInfoItem};
