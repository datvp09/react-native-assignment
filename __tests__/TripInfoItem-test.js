import 'react-native';
import React from 'react';
import {TripInfoItem} from '../src/components';
import renderer from 'react-test-renderer';

describe('TripInfoItem', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(
          <TripInfoItem
            item={{
              trip_id: 17094,
              distance: 16,
              duration: 1956,
              end_time: '2017-07-04T08:23:12+0000',
              start_time: '2017-07-04T07:50:36+0000',
            }}
            style={{}}
          />,
        )
        .toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
