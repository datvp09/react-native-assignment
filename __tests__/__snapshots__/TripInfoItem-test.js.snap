// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripInfoItem Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 7,
        "opacity": 0.9,
        "padding": 10,
        "paddingBottom": 12,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 12,
        }
      }
    >
      TripId: 17094
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 12,
        }
      }
    >
      Distance: 16
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 12,
        }
      }
    >
      Duration: 1956
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 12,
        }
      }
    >
      Start time: 7:50, 4/7/2017
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 12,
        }
      }
    >
      End time: 8:23, 4/7/2017
    </Text>
  </View>
</View>
`;
